docs_src/design/adr/application/0025-Record-and-Replay.md

```diff
@@ -0,0 +1,128 @@
+
+!!! example - "Examples - Replay Rate wait time calculation"
+    Delta time between original Events is 800ms<br/>
+    Replay rate is 2.0 (100% faster) making wait time 400ms (800ms * (1 / 2.0))<br/>
+    Replay rate is 0.5 (100% slower) making wait time 1600ms (800ms * (1 / 0.5))
+
+The replay function will repeat publishing the recorded data per the `Repeat Count` in from the DTO. 
+
+#### Replay Request DTO
+
+##### Replay Rate
+
+Required rate at which to replay the data compared to the rate the data was recorded. Float value greater than 0 where 1 is the same rate, less than 1 is slower rate and greater than 1 is faster rate than the rate the data was recorded. 
+
+##### Repeat Count
+
+Optional count of number of times to repeat the replay. Default is 1 if not specified or value is 0.
+
+### Download endpoint (Export)
+
+This `GET` API will request that the previously recorded data be exported as a file download. It will accept an optional query parameter to specify compression (NONE, ZIP or GZIP). An error is returned if no data has been recorded or invalid compression type requested.
+
+The file content will be the [Recorded Data DTO](#recorded-data-dto) as define below. The request handler will build the DTO described below by extracting the recorded `Events` from in-memory storage, pulling the referenced `Device Profiles` and `Devices` from Core Metadata using the names from in-memory storage. The file extension used will be `.json`, `.zip` or `.gzip` depending on the compression selected.
+
+#### Recorded Data DTO
+
+##### Events
+
+List of `Events` (with `Readings`) that were recorded
+
+##### Device Profiles
+
+List of `Device Profiles` (complete profiles) that are referenced in the recorded `Events` 
+
+##### Devices
+
+List of `Device defintions` that are referenced in the recorded `Events` 
+
+### Upload endpoint (Import)
+
+This `POST` API will upload previously exported recorded data file. It will accept an optional Boolean query parameter to specify to **not** overwrite existing Device Profiles and/or Devices if they already exist. Default is to overwrite existing with those captured with the recorded data.
+
+The request handler will receive the file as a [Recorded Data DTO](#recorded-data-dto) described above and detect if it is compressed and un-compress the contents if needed before un-marshaling the JSON into the DTO. The compression will be determined based on the file extension. The `Event` data from the DTO will then be saved to the in-memory storage along with the Device Profile and Device Names. The `Device Profiles` and `Devices` will be pushed to Core Metadata if they don't exist or if overwrite is enabled.  
+
+!!! note
+    Import will overwrite any previous recorded data.
+
+## Considerations
+- Only one recorded data set is available for replay.
```

Comment:
```suggestion
- Only one recorded data set may be replayed at a time.
```

Reply:
@TomBrennan-Eaton , `may be replayed at a time` infers there may be multiple recorded. Keeping this at crawl phase supporting only a single recording.

Reply:
To me, "available" would mean recordings I had exported. I think I see what you mean, maybe "Only one recorded data set may be imported and held in memory for replay."

Reply:
updated